File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/04.For-Loop/03-additional/11.oddEvenPosition.js
function OddEvenPosition(input) {

    let numCount = Number(input[0]);
    let index = 1;

    let oddSum = 0;
    let oddMin = Number.MAX_SAFE_INTEGER;
    let oddMax = Number.MIN_SAFE_INTEGER;
    let evenSum = 0;
    let evenMin = Number.MAX_SAFE_INTEGER;
    let evenMax = Number.MIN_SAFE_INTEGER;

    for (let i = 1; i <= numCount; i++) {
        let currentNum = Number(input[index++]);
        if (i % 2 !== 0) {
            oddSum += currentNum;
            if (currentNum < oddMin) {
                oddMin = currentNum;
            }
            if (currentNum > oddMax) {
                oddMax = currentNum;
            }
        } else {
            evenSum += currentNum;
            if (currentNum < evenMin) {
                evenMin = currentNum;
            }
            if (currentNum > evenMax) {
                evenMax = currentNum;
            }
        }
    }

    console.log("OddSum=" + oddSum.toFixed(2) + ",");
    if (oddMin === Number.MAX_SAFE_INTEGER) {
        console.log("OddMin=No,");
    } else {
        console.log("OddMin=" + oddMin.toFixed(2) + ",");
    }
    if (oddMax === Number.MIN_SAFE_INTEGER) {
        console.log("OddMax=No,");
    } else {
        console.log("OddMax=" + oddMax.toFixed(2) + ",");
    }
    console.log("EvenSum=" + evenSum.toFixed(2) + ",");
    if (evenMin === Number.MAX_SAFE_INTEGER) {
        console.log("EvenMin=No,");
    } else {
        console.log("EvenMin=" + evenMin.toFixed(2) + ",");
    }
    if (evenMax === Number.MIN_SAFE_INTEGER) {
        console.log("EvenMax=No");
    } else {
        console.log("EvenMax=" + evenMax.toFixed(2));
    }
}
OddEvenPosition(["1", "-5"]);

File: Python_Basics/Homeworks-And-Labs/04.for_loop/03.additional/11.odd_even_position.py
from sys import maxsize

num_count = int(input())

odd_sum = 0
odd_min = maxsize
odd_max = -maxsize
even_sum = 0
even_min = maxsize
even_max = -maxsize

for i in range(1, num_count + 1):
    current_num = float(input())
    if i % 2 != 0:
        odd_sum += current_num
        if current_num < odd_min:
            odd_min = current_num
        if current_num > odd_max:
            odd_max = current_num
    else:
        even_sum += current_num
        if current_num < even_min:
            even_min = current_num
        if current_num > even_max:
            even_max = current_num
            
print(f'OddSum={odd_sum:.2f},')
if odd_min == maxsize:
    print("OddMin=No,")
else:
    print(f"OddMin={odd_min:.2f},")
if odd_max == -maxsize:
    print("OddMax=No,")
else:
    print(f"OddMax={odd_max:.2f},")
print(f"EvenSum={even_sum:.2f},")
if even_min == maxsize:
    print("EvenMin=No,")
else:
    print(f"EvenMin={even_min:.2f},")
if even_max == -maxsize:
    print("EvenMax=No")
else:
    print(f"EvenMax={even_max:.2f}")
